tt_metal/impl/dispatch/kernels/cq_dispatch.cpp

```diff
@@ -526,9 +483,15 @@ void process_write_paged() {
                 dispatch_cb_blocks,
                 dispatch_cb_pages_per_block>(block_noc_writes_to_clear, wr_block_idx);
         }
+        // Transfer size is min(remaining_length, data_available_in_cb)
+        uint32_t available_data = cb_fence - data_ptr;
+        uint32_t xfer_size = (page_size - dst_addr_offset) > available_data ? available_data : (page_size - dst_addr_offset);
```

Comment:
I'd love to know what this does to performance for pages <8K long, has to be a (small?) hit, right?  Can you run moreh's host to device perf test w/ smaller page sizes and measure before/after?

Reply:
I got these results comparing before and after (my results + a couple of small local optimizations I'll push to this PR:
<img width="398" alt="image" src="https://github.com/user-attachments/assets/7aca557f-aa1b-4fab-bcfa-23a780233649">

Results look better as we increase the page size, since it seems like the benefits of kicking of larger writes offsets the cost of additional processing on the dispatcher.

Reply:
We now use one-packet APIs in `process_write_paged`. Results look much better:
![image](https://github.com/user-attachments/assets/960a8c54-dd6b-4d6e-af94-3b9ef0109979)